results and discussion part (#30)
Ctrl-Alt-Defeat-Foundation-Models/informatiCup2024Paper: paper.tex

We used the aforementioned models for generating and detecting texts and images. As for image processors, one was already given in the description of the "InformatiCup2024" \cite{InformatiCup}, the image noising using the Gaussian noise. Image noise is the random variation of brightness or color information in images produced by the sensor and circuitry of a scanner or digital camera. The reason why the Gaussian one, also called normal noise, is being used, is because it is being caused by natural sources and the most common noise that affects images naturally \cite{mohammed2016study}. \\
There are several types of noise that can affect images. Some of these noise models are: Gaussian noise, White noise, Fractal noise, Salt \& Pepper noise, Periodic noise, Quantization noise, Speckle noise, Poisson noise, Poisson-Gaussian noise, Structured noise, Gamma noise and Rayleigh noise \cite{gonzalez2009digital}. The three common types of image noise are: Gaussian noise, Salt \& Pepper noise, and Speckle noise \cite{al2010comparative}. That is why we used those three, as well as the Poisson noise.

\section{Results + Discussion}

> In the following we will have a look at the results of the aforementioned experiment.

Reviewer: ```suggestion
In the following section we take a look on the results of the aforementioned experiment.
```

> In fig. \ref{stable_diffusion} can be seen how the different processors and their combinations effect an image generated by the stable diffusion generator. The bars are shown in packs of three where the most left one represents the nahrawy evaluator, the middle one is resnet18 and lastly on the right is the umm-maybe evaluators respectively. The different colors represent different output. Black simbolizes that the image before processing was detected as AI-generated and after processing as well. Grey shows that both are detected to be human made, whereas for red the process did the exact opposite of what it should do and fool the detector in a way that after processing it is seen as AI-generated whereas before as human made. Lastly the green bar does exactly what is desired. \\

Reviewer: ```suggestion
Fig. \ref{stable_diffusion} shows how the different processors and their combinations effect an image generated by the stable diffusion generator. The bars are structured in packs of three where the most left one represents the nahrawy evaluator, the middle one is resnet18 and on the right is the umm-maybe evaluator respectively. The different colors represent the different outputs. Black symbolizes that the image before and after processing was detected as AI-generated. Grey shows that both are detected to be human made, whereas for red the process did the exact opposite of what it should do and fool the detector in a way that after processing it is seen as AI-generated whereas before as human made. Lastly the green bar does exactly what is desired, so it pictures AI-generated content, where the executed augmentation lead to the detector guessing it was human-made. \\
```

> It can be seen in fig. \ref{fig:stable_diffusion}, that only the nahrawy evaluator has a little bit of red in it but also it is the only one with the output of 'human-human'. Compared to the dall-E generator in fig. \ref{fig:dallE} the amount of green is relatively low. There are also other differences as here only umm-maybe achieved 'human-ai' but also 'human-human'. 

Reviewer: ```suggestion
The data depicted in fig. \ref{fig:stable_diffusion} shows, that only the nahrawy evaluator fooled the detector the wrong way round, but it is also the only one with the output of 'human-human'. Compared to the Dall-E generator in fig. \ref{fig:dallE} the amount of right tricking is relatively low. There are also other differences as here only umm-maybe achieved 'human-ai' but also 'human-human'. 
```

> It is visible that the gaussian noise image processor as well as the salt-and-pepper one are the best processors. This can be seen as they alone achieve a great result as well as their combinations with other ones. It is also worth mentioning that these result are only achieved when using the nahrawy evaluator. Whereas resnet18 which can be seen in both figures has only the output of 'ai-ai'.

Reviewer: ```suggestion
It is visible that the gaussian noise image processor as well as the salt-and-pepper one are the best processors. This can be seen as they solely achieve a great result as well as their combinations with other ones. It is also worth mentioning that these results are only achieved when using the nahrawy evaluator. Whereas resnet18, which can be seen in both figures, couldn't be tricked by any processor.
```

\begin{figure}
	\centering
	\includegraphics[width=\linewidth]{images/stable-diffusion.png}
	\caption{Stable Diffusion with different processors and nahrawy, resnet18 and umm-maybe as evaluators}
	\label{fig:stable_diffusion}
\end{figure}

\begin{figure}
	\centering
	\includegraphics[width=\linewidth]{images/dalle.png}
	\caption{Dall-E with different processors and nahrawy, resnet18 and umm-maybe as evaluators}
	\label{fig:dallE}
\end{figure}

> In fig. \ref{fig:gpt2} the text processors can be seen. The colors stay the same and the concept as well, where the left bar is using the radar evaluator and the right one the roberta base one. It is visible that the double-whitespace processor alone has little effect whereas in combination with the typo one the result rise enourmously. This is due to the typo processor, since it alone also achieve great results. Here is also one evaluator, which produces only a black bar. 

Reviewer: ```suggestion
In fig. \ref{fig:gpt2} the text processors are plotted. The colors stay the same and the concept as well, where the left bar is using the radar evaluator and the right one the roberta base one. It is visible that the double-whitespace processor solely has little effect whereas in combination with the typo processor the result rises enormously. This is due to the typo processor since it alone also achieves great results. The radar evaluator only produces only black bars. 
```

Author: Is 'based' here right ?

> Pretty much the exact same results can be seen in fig. \ref{fig:falcon} where radar has just black bars and typo as well as its combination with the roberta base evaluator achieve great results. The difference is that with the falcon generator and the roberta base evaluator in combination, some texts are detected as human made even before processing, which explains the red and grey parts. 

Reviewer: ```suggestion
Similar results can be seen in fig. \ref{fig:falcon} where radar just contains black bars and typo as well, as its combination with the roberta base evaluator achieves great results. The difference is that with the falcon generator and the roberta base evaluator in combination, some texts are detected as human made even before processing, which explains the red and grey parts. 
```

\begin{figure}
	\centering
	\includegraphics[width=\linewidth]{images/gpt2.png}
	\caption{GPT2 with different processors and nahrawy, resnet18 and umm-maybe as evaluators}
	\label{fig:gpt2}
\end{figure}

\begin{figure}
	\centering
	\includegraphics[width=\linewidth]{images/falcon.png}
	\caption{Falcon with different processors and nahrawy, resnet18 and umm-maybe as evaluators}
	\label{fig:falcon}
\end{figure}

> All in all can be said that the combination between generators and evaluator plays a huge role in the results. In case of the text generators the difference seems minor, but for example just the fact, that the roberta base evaluator detected around ten falcon generated texts as human made in the middle case of fig. \ref{fig:falcon} and after processing as well, which means that if that would not have been the case, we would have a perfect score here. 

Reviewer: ```suggestion
To summarize we can say, that the combination between generators and evaluators plays a huge role in the results. In case of the text generators the difference seems minor, but for example the roberta base evaluator detected around ten falcon generated texts as human made in the middle case of fig. \ref{fig:falcon} and after processing as well, which means that if that would not have been the case, we would have a perfect score here. 
```

> This point is even better visible when comparing the figures \ref{fig:stable_diffusion} \ref{fig:dallE} where the importance of the used generators becomes visible. The stable-diffusion generator produces images where it is difficult to fool a detector whereas when using the same processors and detectors, the images of dallE can be processed in a way to fool some detectors.

Reviewer: ```suggestion
This becomes more clear when comparing fig. \ref{fig:stable_diffusion} with fig \ref{fig:dallE} where the importance of the used generators becomes visible. The stable-diffusion generator produces images where it is difficult to fool a detector whereas when using the same processors and detectors, the images of dallE can be processed in a way to fool some detectors.
```

> Another importance to notice is the usage of different evaluators. In both cases, text and image we had one evaluator respectively which could not be fooled no matter what. These evaluators where so strict, that they would even classify a human made image provided by us as AI-generated.

Reviewer: ```suggestion
Another noticeable result is in the usage of different evaluators. In both cases, text and image, we had one evaluator respectively which could not be fooled with any method. These evaluators where so strict, that they would even classify a human made image provided by us as AI-generated.
```

> If those were the only detectors we would have had, the result would look like there could nothing be done to fool an AI detector.

Reviewer: I think we dn't need this sentence
```suggestion
```
